### PR TITLE
"Add to Steam" and creation of shortcuts

### LINKED ISSUE #358: [Request]: Add individual games/region to Steam
### Is your request related to a problem?

No

### How is the feature request related to the problem?

_No response_

### Describe your proposed solution

Add a mean to add individual game/region to Steam "non-Steam" games, ala Heroic launcher https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher.

## Task list:
- [x] Implement a command parser to go to a region and then launch it
> Make sure it still works when another Collapse is already launched (either redirection or force multi-instance)
- [ ] Implement method to manage requested region into Steam's game library

### Describe alternatives you've considered

Heroic's source code (typescript)
> Add game
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/8bdee1383446d3b81e240a4300baaf337d48ec92/src/backend/shortcuts/nonesteamgame/nonesteamgame.ts#L206

> Remove game
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/8bdee1383446d3b81e240a4300baaf337d48ec92/src/backend/shortcuts/nonesteamgame/nonesteamgame.ts#L387

> Check if added
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/8bdee1383446d3b81e240a4300baaf337d48ec92/src/backend/shortcuts/nonesteamgame/nonesteamgame.ts#L498

### Additional context

Requested by @/thathavonocyanite (310334103041998848) over at Discord https://discord.com/channels/1116150300324139131/1169941928041783337/1169941928041783337

### PR DESCRIPTION
**Pull request by gablm**
#### 1. Shortcut Creation
Asks for a folder and creates a .url file linking to Collapses' URL Protocol. 
"-p" is be added if selected in the UI. 

#### 2. Addition of regions to Steam
Parses and writes a new shortcuts.vdf including a new shortcut for the region for every folder in Steam's userdata folder.
Also downloads the assets related to this shortcut.
If the shortcut already exists, it will be removed and added again while also checking if the md5 checksum for all assets matches and if any file needs to be downloaded.

### Requirements 
 - https://github.com/CollapseLauncher/CollapseLauncher-ReleaseRepo/pull/12
   - Adds the Assets needed to the metadata
 - System.Text.Encoding.CodePages
   - Adds ANSI encoding

closes #358

### CODE REVIEW

**Review by bagusnl — APPROVED** [review]
No code issue other than we have discussed before. 
Requesting recheck by @neon-nyan due to COM interaction and encodings that is beyond my knowledge

File: Hi3Helper.Core/Classes/Shared/Region/LauncherConfig.cs [hunk 1]
```diff
@@ -152,6 +152,12 @@ private static void InitScreenResSettings()
                                                               Name = "Bitbucket",
                                                               URLPrefix = "https://bitbucket.org/neon-nyan/collapselauncher-releaserepo/raw/main",
                                                               Description = Lang._Misc.CDNDescription_Bitbucket
+                                                          }*/
```
Comment (gablm) [inline Hi3Helper.Core/Classes/Shared/Region/LauncherConfig.cs:155]: Note: This needs to be removed before merge/after testing.

**Review by neon-nyan — APPROVED** [review]
Should be looking just fine for me.
Sorry for taking so long to take a look on this

**Comment by bagusnl** [comment]
Approbal achieved

@gablm can you check on the merge conflicts? We can merge it once the conflicts are resolved
Also, you can pick if you want to merge by `Squash` or `Merge commit`

**Comment by gablm** [comment]
There doesn't seem to be any merge conflicts on my end
About merge type, I'd rather if it was by squash

**Comment by bagusnl** [comment]
I was looking at the wrong merge type, rebase causes conflict but squash and commit doesn't :derp:

Alrighty, merging by squash.
Thank you for your work! <3

**Pull request merged**